airflow/providers/cncf/kubernetes/hooks/kubernetes.py

```diff
@@ -96,25 +99,49 @@ def get_ui_field_behaviour() -> Dict:
         }
 
     def __init__(
-        self, conn_id: str = default_conn_name, client_configuration: Optional[client.Configuration] = None
+        self,
+        conn_id: Optional[str] = default_conn_name,
+        client_configuration: Optional[client.Configuration] = None,
+        cluster_context: Optional[str] = None,
+        config_file: Optional[str] = None,
+        in_cluster: Optional[bool] = None,
     ) -> None:
         super().__init__()
         self.conn_id = conn_id
         self.client_configuration = client_configuration
+        self.cluster_context = cluster_context
+        self.config_file = config_file
+        self.in_cluster = in_cluster
+
+    @staticmethod
+    def _coalesce_param(*params):
+        for param in params:
+            if param is not None:
+                return param
 
     def get_conn(self) -> Any:
         """Returns kubernetes api session for use with requests"""
-        connection = self.get_connection(self.conn_id)
-        extras = connection.extra_dejson
-        in_cluster = extras.get("extra__kubernetes__in_cluster") or None
-        kubeconfig_path = extras.get("extra__kubernetes__kube_config_path") or None
+        if self.conn_id:
+            connection = self.get_connection(self.conn_id)
+            extras = connection.extra_dejson
+        else:
+            extras = {}
+        in_cluster = self._coalesce_param(
+            self.in_cluster, extras.get("extra__kubernetes__in_cluster") or None
+        )
```

Comment:
```suggestion
        in_cluster = self.in_cluster or extras.get("extra__kubernetes__in_cluster")
```

Is checking against an explicit None required, or can we just be truthy? Wanting to use a falsy value seems pretty unlikely, no?

Reply:
hmm it's a good question but no i think you if the hook is passed a non-default, i.e. a non-None,  e.g. a `False`, then we should definitely respect that.

example is, the Connection object  has `True` but in our usage we want to override it with `False`.  we should support that and apply the `hook params  trumps connection` rule

Reply:
Oh, I completely overlooked this is a bool 🤦‍♂️. That said, the `or None` here isn't doing anything:

```suggestion
        in_cluster = self._coalesce_param(self.in_cluster, extras.get("extra__kubernetes__in_cluster"))
```

Maybe short enough to be on a single line now too?

Reply:
unfortunately the `or None`  is meaningful and needed.

apparently when you create a connection in the UI, and when that  conn type has these form customizations (which i am really not a huge fan of fwiw), then it creates an empty string value e.g. `extra__kubernetes__in_cluster: ""` in the extra.  

so, we need to get rid of that empty string and replace it with None, and that's what this is doing.

Reply:
Ugh, I remember this now from another PR. I wonder if it is worth a comment.